### PR TITLE
Json

### PR DESCRIPTION
**Pull request by otaviobp**
Some improvements in json formatter I did when I was reworking the progress code.

### CODE REVIEW

File: src/lib/formatter_json.c [hunk 1]
```diff
@@ -28,51 +28,54 @@
 #include "progress.h"
 #include "strings.h"
 
-static int json_format = 0;
```
Comment (castulo) [inline src/lib/formatter_json.c]: are you sure these are not necessary anymore? I don't even remember why we had them in the first place, but I remember they were necessary 🤣

File: src/lib/progress.h [hunk 2]
```diff
@@ -64,26 +66,26 @@ void progress_set_format(progress_fn_t, start_fn_t, end_fn_t);
...
-void progress_init_steps(char *, int);
+void progress_init_steps(const char *title, int steps_in_process);
 
 /**
  * @brief Finishes the reporting of a process (as defined above). If the
  * format used, needs a special closing statement, this function will
  * print it.
  */
-void progress_finish_steps(char *, int status);
+void progress_finish_steps(int status);
```
[the remark this answers is not in the document]
Comment (otaviobp) [reply src/lib/progress.h]: I didn't understand. Comment didn't mention the steps_description that was removed.

[the remark this answers is not in the document]
Comment (castulo) [reply src/lib/progress.h]: well it says: "If the format used, needs a special closing statement, this function will print it", I'm assuming the special closing statement it refers to is the one you provided with the "char *" argument, maybe I'm misunderstanding the comment

File: src/lib/formatter_json.c (hunk 1 above)
Comment (castulo) [reply src/lib/formatter_json.c]: yeah, you are right, and I tested this and looks good... maybe it was needed in earlier versions of the implementation and was never removed. I'm ok with removing this now.

**Pull request merged**